Automattic/jetpack
# Social Previews: Add Modal

Fixes #16196
Fixes #16729
Fixes #16785
Fixes #16805

#### Changes proposed in this Pull Request:
* Adds an actual modal containing social previews that are connected to the current post - author, url, post title, excerpt/content and the featured image…
* Working controls for switching between services
* This PR only implements both the functionality and styling based on the design from the issue

#### Screenshots

| Twitter | Facebook | Google | Upgrade
| - | - | - | -
| <img width="901" alt="Screenshot 2020-08-06 at 18 02 23" src="https://user-images.githubusercontent.com/156676/89554398-18c69d00-d80f-11ea-8d23-7ebeb1e572f9.png"> | <img width="902" alt="Screenshot 2020-08-06 at 18 02 16" src="https://user-images.githubusercontent.com/156676/89554428-21b76e80-d80f-11ea-9c96-dd81ec008116.png"> | <img width="903" alt="Screenshot 2020-08-06 at 18 02 07" src="https://user-images.githubusercontent.com/156676/89554437-24b25f00-d80f-11ea-91c2-5601f38d6d38.png"> | <img width="903" src="https://user-images.githubusercontent.com/156676/89903363-da9ef400-dbe7-11ea-85b6-72392695240a.png" />

#### Jetpack product discussion


#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Same setup as #16615
- Open the Jetpack sidebar and click "Preview"
- A modal window should open with previews
- To test the upgrade modal, you need to manually update the code to include `false` again as it did in 31cdc5d

#### Proposed changelog entry for your changes:
none yet
